src/main/java/space/obminyashka/items_exchange/controller/ImageController.java

```diff
@@ -69,8 +71,12 @@ public List<byte[]> getImagesResource(
     @ResponseStatus(HttpStatus.OK)
     public List<ImageDto> getByAdvertisementId(
             @Parameter(name = "advertisement_id", description = "ID of the Advertisement for getting all the images representation", required = true)
-            @PathVariable("advertisement_id") UUID id) {
-        return imageService.getByAdvertisementId(id);
+            @PathVariable("advertisement_id") UUID id) throws AdvertisementIdNotFoundException {
+        List<ImageDto> listImagesByAdvertisement = imageService.getByAdvertisementId(id);
+        if (listImagesByAdvertisement.isEmpty()) {
+            throw new AdvertisementIdNotFoundException(getMessageSource(ResponseMessagesHandler.ExceptionMessage.ADVERTISEMENT_NOT_EXISTED_ID));
```

Comment:
By following that approach, we will soon get many exceptions with some keywords in it but with absolutely the same behavior: LocationId..., CategoryId..., ImageId... and lots of the same exceptions.
Using a generic one **EntityNotFoundException** with the text that determines its cause would be more beneficial.